### ISSUE TITLE
Wrong query is getting canceled when context is done

### ISSUE THREAD

**Issue by errx** [go-pg/pg#1893]
## Steps to Reproduce
```
package main

import (
	"context"
	"sync"
	"time"

	"github.com/go-pg/pg/v10"
	"github.com/go-pg/pg/v10/orm"
)

type User struct {
	ID   int64
	Name string
}

func target(db *pg.DB) {
	db2 := db.WithContext(context.Background())
	for {
		_, err := db2.Query(new(User), "select *, pg_sleep(0.01) _no from users -- TARGET")
		if err != nil {
			panic(err)
		}
	}
}

func sleepAndCancel(db *pg.DB, wg *sync.WaitGroup) {
	ctx := context.Background()
	for {

		ctx, cancel := context.WithCancel(ctx)
		db2 := db.WithContext(ctx)
		go func() {
			time.Sleep(10 * time.Millisecond) // same as pg_sleep
			cancel()
			// }
		}()
		_, err := db2.Query(new(User), "select *, pg_sleep(0.01) _no from users")
		if err != nil {
			// fmt.Println(err.Error())
			// it's ok
		}
	}
	wg.Done()
}

func main() {
	db := pg.Connect(&pg.Options{
		User:     "...",
		Password: "...",
		Database: "test",
	})
	err := db.Model(new(User)).CreateTable(&orm.CreateTableOptions{
		IfNotExists: true,
	})

	if err != nil {
		panic(err)
	}

	wg := &sync.WaitGroup{}
	for i := 0; i < 20; i++ {
		wg.Add(1)
		go sleepAndCancel(db, wg)
	}

	target(db)
}
```
This program will panic with message:
panic: ERROR #57014 canceling statement due to user request

also in postgres log:
```
2021-07-09 19:12:09.469 MSK [73143 60e87559.11db7 0] ERROR:  canceling statement due to user request

2021-07-09 19:12:09.469 MSK [73143 60e87559.11db7 0] STATEMENT:  select *, pg_sleep(0.01) _no from users -- TARGET  <- this query should't be canceled
```
 

## Detailed Description
Wrong query is getting canceled because underlying connection (same pg's process_id) can be reused by another queries before cancellation request (cancelRequest function).

**Comment by kazhuravlev** [comment]
+1
same problem

**Comment by vmihailenco** [comment]
Thanks for the report. Should be fixed in v10.10.2
